lab2/lab2.md

```diff
@@ -1,30 +1,624 @@
-However, to simplify your design, we have provided you with two different *ports* to memory, an instruction port and a data port.
-These ports are hidden in the `CoreIO` bundle.
-To access the instruction memory, you can use the following code inside your CPU module (e.g., in `cpu.scala`).
+*Turn in via Gradescope*. [See below for details.](#Submission)
 
+# Table of Contents
+
+* [Introduction](#introduction)
+  * [Goals](#goals)
+  * [Chisel](#chisel)
+      * [Using the Singularity image/container](#using-the-singularity-imagecontainer)
+      * [Using Scala, sbt, etc.](#using-scala-sbt-etc)
+  * [Working with the DINO CPU code](#working-with-the-dino-cpu-code)
+
+# Introduction
+
+![Cute Dino](../dino-resources/dino-128.png)
+
+In the last assignemnt you implemented the ALU control and incorporated the ALU Control into the DINO CPU to test some baremetal R-type RISC-V instructions. In this assignement you will implement the branch-control and the main control unit. After implementing the individual components and successfully passing all individual component tests, you will combine these along with the other CPU components to complete the single-cycle DINO CPU.
+
+The simple in-order CPU design is based closely on the CPU model in Patterson and Hennessey's Computer Organization and Design.
+
+
+This is the first time we have used this set of assignments, so there may be mistakes.
+To offset this, we will be offering a variety of ways to get extra credit.
+See [the extra credit page](../extra-credit.md) for an up-to-date list of ways to get extra credit.
+
+
+
+# Part I: Implement the Branch-Control
+
+**The test for this part is `dinocpu.BranchControlTesterLab2`.**
+
+In this part you will be implementing the branch-control component in the CPU design. The branch-control, controls whether or not branches are taken.
+
+It takes four inputs: `branch`, `funct3`, `inputx`, `inputy` and generates one output `taken`.
+
+```
+ 
+branch: true if we are looking at a branch
+funct3: the middle three bits of the instruction (12-14). Specifies the type of branch
+inputx: first value (e.g., reg1)
+inputy: second value (e.g., reg2)
+taken: true if the branch is taken.
+```
+
+The following table details the `funct3` input and which type of branching condition to test:
+
+|      |     |
+|------|-----|
+| 000 | beq  |
+| 001 | bne  |
+| 100 | blt  |
+| 101 | bge  |
+| 110 | bltu |
+| 111 | bgeu |
+
+
+You must take the RISC-V ISA specification and implement the proper control to choose the right type of branch test and correctly set or reset the `taken` output if the branch test passes or fails respectively. 
+You can find the specification in the following places:
+
+
+- [on the RISC-V website](https://riscv.org/specifications/)
+
+|31                 25|24  20|19 15|14     12|11                 7|6       0|
```

Comment:
Something is broken in this table. Maybe this line? I think it's missing a | at the end.